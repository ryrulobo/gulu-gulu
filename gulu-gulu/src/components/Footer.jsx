export default function Footer() {
  const backToTop = () => {
    window.scrollTo({ top: 0, behavior: "smooth" });
  };

  return (
    <div
      className="text-center p-5 border-t border-gray-300"
      data-testid="footer"
    >
      <p>Â© 2022 GuluGulu</p>
      <button className="hover:underline text-blue-700" onClick={backToTop}>
        Back to top
      </button>
    </div>
  );
}
