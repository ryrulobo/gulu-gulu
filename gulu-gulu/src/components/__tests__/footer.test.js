import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";

import Footer from "../Footer";

afterEach(() => {
  cleanup();
});

const backToTop = jest.fn();
global.scrollTo = backToTop;

test("Render footer", () => {
  render(<Footer />);
  const footerElement = screen.getByTestId("footer");
  expect(footerElement).toBeInTheDocument();
  expect(footerElement).toHaveTextContent("Â© 2022 GuluGulu");
  expect(footerElement).toContainHTML("p");
  expect(footerElement).toContainHTML("button");
  const button = screen.getByRole("button");
  fireEvent.click(button);
  expect(backToTop).toHaveBeenCalled();
});

test("Matches snapshots", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
