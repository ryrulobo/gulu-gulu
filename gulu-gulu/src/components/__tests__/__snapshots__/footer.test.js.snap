// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshots 1`] = `
<div
  className="text-center p-5 border-t border-gray-300"
  data-testid="footer"
>
  <p>
    Â© 2022 GuluGulu
  </p>
  <button
    className="hover:underline text-blue-700"
    onClick={[Function]}
  >
    Back to top
  </button>
</div>
`;
